src/main/java/com/google/api/codegen/transformer/go/GoSurfaceNamer.java

```diff
@@ -0,0 +1,196 @@
+import com.google.api.tools.framework.model.TypeRef;
+
+import io.grpc.Status;
+
+import java.io.File;
+import java.util.List;
+
+public class GoSurfaceNamer extends SurfaceNamer {
+
+  private final GoModelTypeNameConverter converter;
+  private final Model model;
+  private final String packagePath;
+
+  public GoSurfaceNamer(Model model, String packagePath) {
+    this(new GoModelTypeNameConverter(), model, packagePath);
+  }
+
+  private GoSurfaceNamer(GoModelTypeNameConverter converter, Model model, String packagePath) {
+    super(new GoNameFormatter(), new ModelTypeFormatterImpl(converter), new GoTypeTable());
+    this.converter = converter;
+    this.model = model;
+    this.packagePath = packagePath;
+  }
+
+  @Override
+  public String getPathTemplateName(Interface service, CollectionConfig collectionConfig) {
+    return inittedConstantName(
+        Name.from(
+            getReducedServiceName(service), collectionConfig.getEntityName(), "path", "template"));
+  }
+
+  @Override
+  public String getPathTemplateNameGetter(Interface service, CollectionConfig collectionConfig) {
+    return methodName(
+        Name.from(getReducedServiceName(service), collectionConfig.getEntityName(), "path"));
+  }
+
+  @Override
+  public String getStaticLangReturnTypeName(Method method, MethodConfig methodConfig) {
+    return converter.getTypeName(method.getOutputType()).getFullName();
+  }
+
+  @Override
+  public List<String> getDocLines(ProtoElement element) {
+    if (!(element instanceof Method)) {
+      return super.getDocLines(element);
+    }
+    Method method = (Method) element;
+    String text = DocumentationUtil.getDescription(method);
+    text = lowerFirstLetter(text);
+    return super.getDocLines(getApiMethodName(method) + " " + text);
+  }
+
+  @Override
+  public String getAndSavePagedResponseTypeName(ModelTypeTable typeTable, TypeRef resourceType) {
+    String typeName = converter.getTypeNameForElementType(resourceType).getNickname();
+    int dotIndex = typeName.indexOf('.');
+    if (dotIndex >= 0) {
+      typeName = typeName.substring(dotIndex + 1);
+    }
+    return Name.anyCamel(typeName).join("iterator").toUpperCamel();
+  }
+
+  private static String lowerFirstLetter(String s) {
+    if (s.length() > 0) {
+      s = Character.toLowerCase(s.charAt(0)) + s.substring(1);
+    }
+    return s;
+  }
+
+  private static String upperFirstLetter(String s) {
+    if (s.length() > 0) {
+      s = Character.toUpperCase(s.charAt(0)) + s.substring(1);
+    }
+    return s;
+  }
+
+  @Override
+  public String getGrpcClientTypeName(Interface service) {
+    return converter.getTypeName(service).getNickname() + "Client";
+  }
+
+  @Override
+  public String getGrpcClientConstructorName(Interface service) {
+    return getGrpcClientTypeName(service).replace(".", ".New");
+  }
+
+  @Override
+  public String getCallSettingsTypeName(Interface service) {
+    return clientNamePrefix(service).join("call").join("options").toUpperCamel();
```

Comment:
use className(...) instead of toUpperCamel() so that this code is more "copyable". Same with the other functions in this file that use toUpperCamel() for a type name.